### PR TITLE
Us68 modify mortgage deed and borrower details

### PR DESCRIPTION
**Pull request by danielmarker**
Dev work completed. Swagger updated and changes pulled through from middle name bug fix. Please review. 
**_DONT MERGE AS NO ACCEPTANCE TESTS**_

### CODE REVIEW

File: application/deed/views.py [hunk 1]
```diff
@@ -97,3 +97,73 @@ def delete_borrower(borrower_id):
...
+    error_count, error_message = validate_helper(new_deed_json)
+
+    if error_count > 0:
+        return error_message, status.HTTP_400_BAD_REQUEST
+    else:
+        # If Valid:
+        # Get Current Deed
+        result = Deed.query.filter_by(token=str(deed_reference)).first()
+        existing_deed_borrowers = result.deed['borrowers']
+        i = 0
```
Comment (endamccormack) [inline application/deed/views.py]: better to use something more like this http://scribu.net/blog/python-equivalents-to-phps-foreach.html

File: application/borrower/server.py [hunk 2]
```diff
@@ -2,8 +2,11 @@
 
 
 class BorrowerService:
-    def saveBorrower(self, borrower):
+    def saveBorrower(self, borrower, id=None):
```
Comment (endamccormack) [inline application/borrower/server.py:5]: Wont this be None anyway by default

File: application/deed/views.py [hunk 3]
```diff
@@ -97,3 +97,73 @@ def delete_borrower(borrower_id):
         abort(status.HTTP_404_NOT_FOUND)
     else:
         return json.dumps({'id': borrower_id}), status.HTTP_200_OK
+
+
+@deed_bp.route('/<deed_reference>', methods=['PUT'])
+def get_existing_deed_and_update(deed_reference):
+
+    # Firstly check payload coming in is valid:
+    new_deed_json = request.get_json()
```
Comment (endamccormack) [inline application/deed/views.py]: updated_deed_json\* ?

File: application/deed/views.py [hunk 4]
```diff
@@ -97,3 +97,73 @@ def delete_borrower(borrower_id):
...
+    borrowerService = BorrowerService()
+
+    error_count, error_message = validate_helper(new_deed_json)
+
+    if error_count > 0:
+        return error_message, status.HTTP_400_BAD_REQUEST
+    else:
+        # If Valid:
```
Comment (endamccormack) [inline application/deed/views.py]: I dont think this should be a 2 line comment

File: application/deed/views.py [hunk 5]
```diff
@@ -97,3 +97,73 @@ def delete_borrower(borrower_id):
...
+        existing_deed_borrowers = result.deed['borrowers']
+        i = 0
+        for existing_borrower in existing_deed_borrowers:
+            new_deed_json['borrowers'][i]['id'] = existing_borrower['id']
+            i = i + 1
+
+    if result is None:
+        abort(status.HTTP_404_NOT_FOUND)
+    else:
+
```
Comment (endamccormack) [inline application/deed/views.py]: No need for this new line

File: application/borrower/server.py (hunk 2 above)
Comment (danielmarker) [inline application/borrower/server.py:5]: This is for when a id is passed in it will update instead of assign a new one.

**Comment by endamccormack** [comment]
This is nearly complete, needs acceptance tests and to fix unit tests however we are parking it for now and picking it up at a later date

**Pull request closed without merging**